src/sasl-external.js

```diff
@@ -14,7 +14,7 @@ export default class SASLExternal extends SASLMechanism {
         super(mechname, isClientFirst, priority);
     }
 
-    onChallenge (connection) { // eslint-disable-line class-methods-use-this
+    async onChallenge (connection) { // eslint-disable-line class-methods-use-this
```

Comment:
If you use `await` before `onChallenge` as suggested elsewhere, then there's no need to explicitly make every `onChallenge` function `async`.